File: java/src/main/java/mx/iteso/Priviledge/Publish.java
package mx.iteso.Priviledge;

import mx.iteso.PriviledgeDecorator;
import mx.iteso.User;

public class Publish extends PriviledgeDecorator {
    User user;

    public Publish (User user){
        this.user = user;
    }

    @Override
    public String getDescription() {
        return user.getDescription() + " can publish";
    }
}

File: csharp/decorator-practice/decorator-practice/Publish.cs
namespace decorator_practice
{
    internal class Publish : PrivilegesDecorator
    {
        private readonly User _user;

       
        public Publish(User userIn)
        {
            _user = userIn;
        }

        override
        public string PublishM()
        {
            return "Can publish";
        }

        override
        public string DeleteM()
        {
            return _user.DeleteM();
        }

        override
        public string ApproveM()
        {
            return _user.ApproveM();
        }

        override
        public string CommentM()
        {
            return _user.CommentM();
        }
    }
}
